Re-assess GSEA results at tumor purity threshold

sjspielman opened the pull request:
Closes #1626 

This PR updates the GSEA module to assess to what extent key results (indicated in issue #1626) might change if we perform GSEA only on tumors that pass a purity threshold.

Punchline: Results are almost entirely consistent ✅ 


There are several key code changes:
- Added an flag in the module crux script to generate results from thresholded data 
- Added a notebook to process these results and compare them back to the original results. This notebook is quite granular and walks through _each_ comparison we emphasize in the manuscript. While there is some (not surprising) loss of power in certain comparisons, broadly speaking results are consistent! This notebook can therefore serve as a formal record of the result consistency. Here's the rendered version: 
[03-assess-gsea-at-threshold.nb.html.zip](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/10510714/03-assess-gsea-at-threshold.nb.html.zip)
- Both of the above are now in the module run script and documented in `README` which was amusingly out of date
- There are various HTML diffs to the 2nd notebook from re-running (I wonder if it was not originally generated in docker, which would have been my bad!)
- Some tiny recreational revisions......
   - I removed a print statement
   - Updated the significance threshold in the 02 notebook to be 0.05 which is what we use in the MS, but results presented in the notebooks do not change! The result files got slightly tweaked because they're binary, and the TUKEY results have a column indicating significance so that column may be a tad different. But, it's now more consistent with the MS which I think is probably slightly better.

jaclyn-taroni reviewed (commented):
I have some issues with how we're talking about the specific comparisons in parts 1 & 2 (provided I'm correct about what's being counted, of course). Frankly, I don't find the language in the manuscript to be the most precise either (as compared to what's being assessed in part 3), so what you _did_ seems reasonable. It's the description I'm bothered by (again, if I'm correct 😄 ).

This discussion is going to impact much of the text in the notebook, so returning the review now.

sjspielman commented on analyses/gene-set-enrichment-analysis/01-conduct-gsea-analysis.R:
```diff
@@ -83,6 +87,38 @@ expression_data <- as.data.frame( readr::read_rds(expression_data_file) )
+    dplyr::pull(Kids_First_Biospecimen_ID) %>%
+    unique()
+  
+  # To avoid CI errors, first subset `bs_ids`` to the intersection with `expression_data` colnames
```
EXTEND was a matrix!

sjspielman replied:
Noting this review will get outdated once #1663 is merged.

sjspielman commented on analyses/gene-set-enrichment-analysis/01-conduct-gsea-analysis.R:
```diff
@@ -83,6 +87,38 @@ expression_data <- as.data.frame( readr::read_rds(expression_data_file) )
+  # Subset the `expression_data` variable to those IDs
+  expression_data <- expression_data[, bs_ids]
+  
+  # Quick check on filtering:
+  if (!length(bs_ids) == ncol(expression_data)) {
+    stop("Error filtering IDs to use in GSEA calculation with tumor purity thresholding on.")
+  }
```
Noting this review will get outdated once #1663 is merged.

sjspielman commented:
I addressed reviewer comments including:
- Not quoting things that aren't quotes, given that there are quotes!
- Doc updates and argument name updates for clarity
- Comparisons across a given set of pathways, not pathways themselves, are up/downregulated.

In https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1659/commits/39cdbc132832b63bc0f8a844dacc36c9402973e8, I incorporated the tumor purity filtering function and re-ran the module. There were no diffs from re-running.

jaclyn-taroni commented on analyses/gene-set-enrichment-analysis/run-gsea.sh:
```diff
@@ -41,8 +41,27 @@ INPUT_FILE="${DATA_DIR}/pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds"
 fi
+
+
+######## Calculate scores from stranded expression data for threshold-passing tumors only #######
+INPUT_FILE="${DATA_DIR}/pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds"
+OUTPUT_FILE="${RESULTS_DIR}/gsva_scores_stranded_thresholded.tsv"
+Rscript --vanilla 01-conduct-gsea-analysis.R --input ${INPUT_FILE} --output ${OUTPUT_FILE} --apply_tumor_purity_threshold
+
+
+# Assess results generated for threshold-passing tumors
+Rscript -e "rmarkdown::render('03-assess-gsea-at-threshold.Rmd', clean = TRUE)"
```
This requires modeling that relies on having cancer group labels in place, correct? Wouldn't we want this to be within the same logic as `02-model-gsea.Rmd` above in that case?

sjspielman replied:
I'm not sure I follow what you mean by "cancer group labels in place"? The `03-assess-gsea-at-threshold.Rmd` notebook actually performs its relevant modeling directly rather than reading in anything that would have been produced from `02-model-gsea.Rmd`. The main reason I made that choice is because we want this analysis to focus on the grouping `cancer_group_display`, not `cancer_group`, and I didn't want to add all kinds of wrangling to change that out in the `02-model-gsea.Rmd` notebook.

jaclyn-taroni replied:
Right, but how an individual sample gets grouped into `cancer_group_display` is based on it's `cancer_group` label. A `cancer_group` label is subject to change during the subtyping process. So if you have an analysis that's in any way reliant on the `cancer_group` label (which this is), you'd only want that part to run when you are not subtyping.

sjspielman replied:
In https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1659/commits/7994f234ad427fb63e70b7feb143f551ea56ea07 I updated the logic in `run-gsea.sh` accordingly to now only run `cancer_group`-consuming items if we are NOT subtyping.

In https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1659/commits/7994f234ad427fb63e70b7feb143f551ea56ea07 and https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1659/commits/b909aaea43badf14f19bc95d527965bce19de8aa, I cleaned up some related comments in that `run-gsea.sh` script as well.

jaclyn-taroni reviewed (approved):
We should get the logic worked out, but I think enumerating or hashing out any remaining concerns about this analysis (or more specifically, how we talk about the analysis in the manuscript) would be misplaced here!
